components/org.wso2.carbon.siddhi.editor.core/src/main/java/org/wso2/carbon/siddhi/editor/core/internal/EditorMicroservice.java

```diff
@@ -844,6 +848,53 @@ public Response getAttributes(@PathParam("siddhiAppName") String siddhiAppName,
                 ).build();
     }
 
+    /**
+     * Converts a given Siddhi App string to a specific JSON format for a graph that diagrammatically
+     * display's the Siddhi App to be generated in the Editor design view.
+     *
+     * @param siddhiAppBase64 The Siddhi App (encoded to Base64) to be converted to JSON
+     * @return The JSON result in a predefined format
+     */
+    @POST
+    @Path("/event-flow")
+    @Consumes(MediaType.APPLICATION_FORM_URLENCODED)
+    @Produces(MediaType.APPLICATION_JSON)
+    public Response constructEventFlowJsonString(String siddhiAppBase64) {
+        Response response;
+
+        try {
+            String siddhiAppString = new String(Base64.getDecoder().decode(siddhiAppBase64),
+                    Charset.defaultCharset());
+
+            SiddhiAppMap siddhiAppMap = new SiddhiAppMap(siddhiAppString);
+            EventFlow eventFlow = new EventFlow(siddhiAppMap);
+
+            // The 'Access-Control-Allow-Origin' header must be set to '*' as this might be accessed
+            // by other domains in the future.
+            response = Response.status(Response.Status.OK)
+                    .header("Access-Control-Allow-Origin", "*")
+                    .entity(eventFlow.getEventFlowJSON().toString())
+                    .build();
+        } catch (SiddhiAppCreationException e) {
+            // SiddhiAppCreationException is caught if the provided 'SiddhiAppString' contains any syntax errors.
+            log.info("Cannot Generate Graph View: " + e.getMessage());
+            response = Response.status(Response.Status.BAD_REQUEST)
+                    .header("Access-Control-Allow-Origin", "*")
+                    .entity(e.getMessage())
+                    .build();
+        } catch (Throwable e) {
+            // Throwable is caught, if any unknown errors occur when executing the code in the try block.
```

Comment:
Why `Exception` is not enough? Does a Siddhi app might throw an `Error`?

Reply:
Initially, `Exception` was caught, but it was requested to be changed to `Throwable`.

Reply:
That is not a sufficient answer :smile: As the code author you should know what is the underline reason behind _this part is done this way instead of that way_ in your code

Reply:
Noted. Will change to `Exception`.

Reply:
IMHO, unless we have a really good reason catching `Exception` is fine here. Read more: https://stackoverflow.com/a/6083271/1577286